flink-core/src/main/java/org/apache/flink/api/common/functions/util/AbstractRuntimeUDFContext.java

```diff
@@ -175,6 +177,11 @@ public DistributedCache getDistributedCache() {
 		return this.distributedCache;
 	}
 
+	@Override
+	public <T extends ExternalResourceInfo> Set<T> getExternalResourceInfos(String resourceName, Class<T> externalResourceType) {
+		throw new UnsupportedOperationException("Do not support external resource in current environment");
+	}
+
```

Comment:
The `RuntimeUDFContext` and its subclass is used for [Collection Environment](https://ci.apache.org/projects/flink/flink-docs-master/dev/local_execution.html#collection-environment). As the docs say:

> Typical use-cases for this mode are automated tests, debugging and code re-use.

This mode will not start the full Flink runtime, the task executor will not be started. Since currently the external resource is a TM level component. From my side, it probably not make sense to support it in this environment.